src/test/java/roomescape/application/reservation/ReservationServiceTest.java

```diff
@@ -0,0 +1,279 @@
+        CreateReservationParam param = new CreateReservationParam(LocalDate.now(clock), time.getId(), invalidThemeId,
+                member.getId());
+
+        // when
+        // then
+        assertThatThrownBy(() -> reservationService.create(param))
+                .isInstanceOf(NotFoundEntityException.class)
+                .hasMessage("999에 해당하는 theme 튜플이 없습니다.");
+    }
+
+    @Test
+    void 같은_날짜와_같은_시간과_같은_테마에_예약이_존재한다면_예약을_생성하면_예외가_발생한다() {
+        // given
+        Member member = memberRepository.save(new Member("벨로", new Email("test@email.com"), "pw", Role.NORMAL));
+        Theme theme = themeRepository.save(new Theme("테마", "설명", "이미지"));
+        ReservationTime time = reservationTimeRepository.save(new ReservationTime(LocalTime.of(13, 0)));
+        reservationRepository.save(new Reservation(member, LocalDate.now(clock), time, theme));
+        CreateReservationParam param = new CreateReservationParam(
+                LocalDate.now(clock),
+                time.getId(),
+                theme.getId(),
+                member.getId()
+        );
+
+        // when
+        // then
+        assertThatThrownBy(() -> reservationService.create(param))
+                .isInstanceOf(BusinessRuleViolationException.class)
+                .hasMessage("날짜와 시간이 중복된 예약이 존재합니다.");
+    }
+
+    @Test
+    void 예약을_삭제할_수_있다() {
+        // given
+        Member member = memberRepository.save(new Member("벨로", new Email("test@email.com"), "pw", Role.NORMAL));
+        Theme theme = themeRepository.save(new Theme("테마", "설명", "이미지"));
+        ReservationTime time = reservationTimeRepository.save(new ReservationTime(LocalTime.of(13, 0)));
+        Reservation reservation = reservationRepository.save(
+                new Reservation(member, LocalDate.now(clock), time, theme));
+
+        // when
+        reservationService.deleteById(reservation.getId());
+
+        // then
+        assertThat(reservationRepository.findById(reservation.getId())).isNotPresent();
+    }
+
+    @Test
+    void 전체_예약을_조회할_수_있다() {
+        // given
+        Member member = memberRepository.save(new Member("벨로", new Email("test@email.com"), "pw", Role.NORMAL));
+        Theme theme = themeRepository.save(new Theme("테마", "설명", "이미지"));
+        ReservationTime time1 = reservationTimeRepository.save(new ReservationTime(LocalTime.of(13, 0)));
+        ReservationTime time2 = reservationTimeRepository.save(new ReservationTime(LocalTime.of(14, 0)));
+        reservationRepository.save(new Reservation(member, LocalDate.now(clock), time1, theme));
+        reservationRepository.save(new Reservation(member, LocalDate.now(clock), time2, theme));
+
+        // when
+        List<ReservationResult> reservationResults = reservationService.findAll();
+
+        // then
+        assertThat(reservationResults)
+                .isEqualTo(List.of(
+                                new ReservationResult(
+                                        1L,
```

Comment:
아래와 같이 작성하는 것이 더 정확할 것 같네요 😄 
```suggestion
                                        time1.getId(),
```

Reply:
말씀해주신 것처럼 `1L` 같은 ID를 하드코딩하는 것보다는,
`time1.getId()`처럼 직접 조회한 값을 사용하는 방식이
유지보수 측면에서 더 유연하고 안전하다고 생각합니다. 👍 

다만, 이 테스트에서는 `AbstractServiceIntegrationTest#resetAutoIncrement()`로 각 테스트 메서드 실행 전에 (AutoIncreament로 올라간)ID를 매번 초기화하고 있으며, `@DataJpaTest`에 포함된 `@Transactional` 덕분에 테스트 간 독립적인 DB 상태가 보장된다고 판단했습니다.
해당 환경에서는 1L, 2L과 같은 명시적인 ID 값이 오히려 테스트 데이터를 의도적으로 확인할 수 있는 좋은 단서가 될 수 있다고 생각했습니다.

또한, 만약 이 값이 어긋난다면 오히려 초기화 로직이 제대로 동작하지 않음을 테스트 실패로 드러낼 수 있는 효과도 기대했습니다.

이 테스트는 독립된 환경에서 서비스 메서드의 동작을 검증하는 것이 목표였기 때문에, 현재 방식이 그 목적을 명확히 보여줄 수 있다고 판단했습니다.

물론, 말씀해주신 것처럼 ID 생성 정책이 바뀌거나,
DB 초기화 방식이 달라진다면 이 테스트는 깨질 수 있습니다.
그 경우에는 테스트하고자 하는 범위를 다시 합의하거나
제안해주신 방식으로 전환하는 것이 더 바람직하다고 생각합니다.

이와 관련하여 엘리의 경험이나 생각이 궁금합니다!! 😊

Reply:
> 해당 환경에서는 1L, 2L과 같은 명시적인 ID 값이 오히려 테스트 데이터를 의도적으로 확인할 수 있는 좋은 단서가 될 수 있다고 생각했습니다. 또한, 만약 이 값이 어긋난다면 오히려 초기화 로직이 제대로 동작하지 않음을 테스트 실패로 드러낼 수 있는 효과도 기대했습니다.

맞습니다.
하지만 벨로가 `초기화 동작까지 확인한다`는 의도를 갖고 테스트를 작성했을지라도, 다른 사람이 해당 테스트만 보고 그 의도까지 파악할 수 있을까요? (일단 저는 파악하지 못했습니다!)
초기화 로직을 검증하기 위한 별도 테스트를 구성하는 것이 협업의 측면에선 더 명확한 코드일 것 같네요 😄